strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/storage/metadata/MavenSnapshotManager.java

```diff
@@ -205,13 +205,17 @@ private boolean removeTimestampedSnapshot(RepositoryPath basePath,
                                   }
                               });
         }
-        else if (numberToKeep == 0 && keepPeriod != 0)
+        else if (numberToKeep == 0 && keepDate != null)
         {
             snapshots.forEach((k, v) ->
                               {
                                   try
                                   {
-                                      if (keepPeriod < getDifferenceDays(v.getTimestamp()))
+                                      DateFormat formatter = new SimpleDateFormat(TIMESTAMP_FORMAT);
+                                      Calendar calendar = Calendar.getInstance();
```

Comment:
we should use `v.getTimestamp()` here instead of current date from `Calendar`:
```
Date snapshotVersionDate = formatter.parse(buildTimestamp);
```

and also `getDifferenceDays` method should be removed